test/integration/acceptance/query_permission_test_txs.cpp

```diff
@@ -0,0 +1,88 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "query_permission_test_txs.hpp"
+
+QueryPermissionTxs::QueryPermissionTxs()
+    : QueryPermissionTestBase({Role::kGetMyAccTxs},
+                              {Role::kGetDomainAccTxs},
+                              {Role::kGetAllAccTxs}) {}
+
+IntegrationTestFramework &QueryPermissionTxs::prepareState(
+    AcceptanceFixture *fixture,
+    const interface::RolePermissionSet &spectator_permissions /*= {}*/) {
+  auto target_permissions = spectator_permissions;
+  target_permissions.set(Role::kReceive);
+  target_permissions.set(Role::kTransfer);
+  target_permissions.set(Role::kAddAssetQty);
+  target_permissions.set(Role::kSubtractAssetQty);
+  target_permissions.set(Role::kSetDetail);
+
+  // Make some transactions of different kinds:
+  std::vector<shared_model::proto::Transaction> user_transactions{
+      // Add asset to user
+      fixture->complete(
+          fixture->baseTx().addAssetQuantity(kAssetId, "20000.0")),
+      // Transfer assets to admin
+      fixture->complete(fixture->baseTx().transferAsset(
+          kUserId, kAdminId, kAssetId, "outgoing", "500.0")),
+      // Set account details
+      fixture->complete(fixture->baseTx(kUserId).setAccountDetail(
+          kUserId, "key1", "val1"))
+  };
+
+  std::transform(user_transactions.cbegin(),
+                 user_transactions.cend(),
+                 std::back_inserter(tx_hashes_),
+                 [](const auto &tx) { return tx.hash(); });
+
+  // initial state setup
+  auto &itf = QueryPermissionTestBase::prepareState(
+      fixture, spectator_permissions, target_permissions);
+
+  for (const auto& tx : user_transactions) {
+    itf.sendTxAwait(tx, getBlockTransactionsAmountChecker(1));
+  }
+
+  return itf;
+}
+
+std::function<void(const proto::QueryResponse &response)>
+QueryPermissionTxs::getGeneralResponseChecker() {
+  return [this](const proto::QueryResponse &response) {
+    ASSERT_NO_THROW({
+      const auto &resp = boost::apply_visitor(
+          framework::SpecifiedVisitor<interface::TransactionsResponse>(),
+          response.get());
+
+      const auto &transactions = resp.transactions();
+      ASSERT_EQ(boost::size(transactions), tx_hashes_.size());
+      std::vector<shared_model::interface::types::HashType> resp_tx_hashes;
+      resp_tx_hashes.reserve(tx_hashes_.size());
+      std::transform(resp.transactions().begin(),
+                     resp.transactions().end(),
+                     std::back_inserter(resp_tx_hashes),
+                     [](const shared_model::interface::Transaction &tx) {
+                       return tx.hash();
+                     });
+      for (const auto &tx_hash : tx_hashes_) {
+        ASSERT_NE(
+            std::find(resp_tx_hashes.cbegin(), resp_tx_hashes.cend(), tx_hash),
+            resp_tx_hashes.cend());
```

Comment:
That is quite tough to analyze in case of fail. So it might be better write to output some useful info like tx hash